watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1595_photothermal_membrane_candoP/amo_1595.py

```diff
@@ -0,0 +1,330 @@
+#
+###############################################################################
+
+import os
+
+from pyomo.environ import (
+    ConcreteModel,
+    units as pyunits,
+    assert_optimal_termination,
+    Expression,
+    value,
+    TransformationFactory,
+)
+
+from pyomo.network import Arc, SequentialDecomposition
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock
+from idaes.core.util import get_solver
+from idaes.generic_models.unit_models import Product
+import idaes.core.util.scaling as iscale
+from idaes.generic_models.costing import UnitModelCostingBlock
+
+from watertap.core.util.initialization import assert_degrees_of_freedom
+
+from watertap.core.wt_database import Database
+import watertap.core.zero_order_properties as prop_ZO
+from watertap.unit_models.zero_order import (
+    FeedZO,
+    PhotothermalMembraneZO,
+    CANDOPZO,
+    PumpElectricityZO,
+)
+from watertap.core.zero_order_costing import ZeroOrderCosting
+
+
+def main():
+    m = build()
+
+    set_operating_conditions(m)
+    assert_degrees_of_freedom(m, 0)
+    assert_units_consistent(m)
+
+    initialize_system(m)
+
+    results = solve(m)
+    assert_optimal_termination(results)
+    display_results(m)
+
+    add_costing(m)
+    m.fs.costing.initialize()
+
+    assert_degrees_of_freedom(m, 0)
+    assert_units_consistent(m)
+    results = solve(m)
+    assert_optimal_termination(results)
+    display_costing(m)
+
+    return m, results
+
+
+def build():
+    # flowsheet set up
+    m = ConcreteModel()
+    m.db = Database()
+
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    m.fs.prop = prop_ZO.WaterParameterBlock(
+        default={
+            "solute_list": [
+                "nitrogen",
+                "phosphates",
+                "bioconcentrated_phosphorous",
+                "nitrous_oxide",
+            ]
+        }
+    )
+
+    # unit models
+    # feed
+    m.fs.feed = FeedZO(default={"property_package": m.fs.prop})
+
+    # pump
+    m.fs.pump = PumpElectricityZO(
+        default={
+            "property_package": m.fs.prop,
+            "database": m.db,
+        },
+    )
+
+    # photothermal membrane
+    m.fs.photothermal = PhotothermalMembraneZO(
+        default={
+            "property_package": m.fs.prop,
+            "database": m.db,
+        },
+    )
+
+    # CANDO+P reactor
+    # Note that CANDOPZO model electricity costing includes pumping costs for
+    # the unit, so a pump has not been included in this flowsheet between the
+    # photothermal membrane and CANDO+P units
+    m.fs.candop = CANDOPZO(
+        default={
+            "property_package": m.fs.prop,
+            "database": m.db,
+        },
+    )
+
+    # product streams
+    m.fs.photothermal_water = Product(default={"property_package": m.fs.prop})
+    m.fs.candop_treated = Product(default={"property_package": m.fs.prop})
+    m.fs.candop_byproduct = Product(default={"property_package": m.fs.prop})
```

Comment:
What are the actual products of `candop_treated` and `candop_byproduct`? I am not sure but if there are no requirements or limitations on the `candop_byproduct` term, you probably don't need to add this.

Reply:
The CANDO+P byproduct stream is just N2O. The treated stream is everything else - H2O, bioconcentrated phosphorous (BCP), nitrogen, and phosphates.

Reply:
You can see in this flow diagram that the N2O and BCP streams are supposed to be separate:

![image](https://user-images.githubusercontent.com/26798258/174652021-815b1672-aa0d-4a28-b0c8-c471b9133148.png)

Reply:
@adam-a-a suggested that I could consider the price of N2O as one of the costing parameters, and factor the dollars recouped by selling that product into the levelized cost calculations. But I opted not to do that, because as you can see in the flow diagram, the N2O is just passed to the combustion block to generate electricity.

Reply:
Yeah, that makes sense. The N2O can pass to the combustion for energy recovery.

Reply:
@adam-a-a I'm not sure it makes sense to consider an equivalent amount of electricity for the N2O, since N2O isn't directly flammable. From [wiki](https://en.wikipedia.org/wiki/Nitrous_oxide#:~:text=Internal%20combustion%20engine,-Main%20article%3A%20Nitrous&text=In%20vehicle%20racing%2C%20nitrous%20oxide,to%20produce%20more%20engine%20power.), the way it works in engines is that the N2O "allows the engine to burn more fuel by providing more oxygen during combustion. The increase in oxygen allows for an increase in the injection of fuel, allowing the engine to produce more engine power. The gas is not flammable at a low pressure/temperature, but it delivers more oxygen than atmospheric air by breaking down at elevated temperatures, about 570 degrees F (~300C)." I assume the principle is the same here. My inclination is to assign a proxy price to the N2O, which I can do.